lib/ios-log.js

```diff
@@ -0,0 +1,235 @@
+      } else {
+        logger.warn('Could not capture device log using libimobiledevice idevicesyslog.' +
+                    ' Libimobiledevice probably is not installed');
+        logger.debug('Attempting iOS device log capture via deviceconsole');
+        let limDir = path.resolve(__dirname, '../../../build/deviceconsole');  //??? 
+        spawnEnv.PATH = `${process.env.PATH}:${limDir}`;
+        spawnEnv.DYLD_LIBRARY_PATH = limDir + ':' + process.env.DYLD_LIBRARY_PATH;
+        this.proc = spawn('deviceconsole', ['-u', this.udid], {env: spawnEnv});
+      }
+      this.finishStartingLogCapture();
+    } else {
+      let xcodeVersion = await xcode.getVersion();
+      let ver = parseInt(xcodeVersion.split('.')[0], 10);
+      if (ver >= 5) {
+        let logsPath;
+        if (ver >= 6) {
+          logger.debug('Starting iOS 8.* simulator log capture');
+          if (_.isUndefined(this.simUdid)) {
+            logger.errorAndThrow('ios8 log capture requires a sim udid'); 
+          }
+          logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'CoreSimulator', this.simUdid); 
+        } else {
+          logger.debug('Starting iOS7.* simulator log capture');
+          logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'iOS Simulator', '7.*');
+        }
+
+        let touchLog = async function () {
+          if (logsPath.indexOf('*') >= 0) {
+            return; 
+          }
+          await mkdirp(logsPath);
+          await touch(path.resolve(logsPath, 'system.log'));
+          await fs.appendFile(path.resolve(logsPath, 'system.log'), 
+                          'A new Simulator is about to start!'); 
+        };
+        let getLogFiles = async function () {
+          try {
+            let files = await glob(path.resolve(logsPath, 'system.log'));
+            if (files.length < 1) {
+              throw 'Could not start log capture';
+            }
+          } catch (e) {
+            logger.error('Could not start log capture because no iOS ' +
+                'simulator logs could be found at ' + logsPath + '/system.log' +
+                'Logging will not be functional for this run');
+          }
+        };
+        let tailLogs = function (files) {
+          let lastModifiedLogPath = files[0];
+          let lastModifiedLogTime = fs.statSync(files[0]).mtime; //promisify this 
+          for (let file in files) {
+            let mtime = fs.statSync(file).mtime;
+            if (mtime > lastModifiedLogTime) {
+              lastModifiedLogPath = file;
+              lastModifiedLogTime = mtime;
+            }
+          }
+          this.proc = spawn('tail', ['-f', '-n', '1', lastModifiedLogPath]);
+          this.finishStartingLogCapture();
+        };
+        try {
+          await touchLog();
+          await getLogFiles();
+          await tailLogs();
+        } catch (err) {
+          logger.warn('System log capture failed.');
+          return;
+        }
+      } else {
```

Comment:
here's a pattern that I like to use when I have long conditionals is. the way the code is currently structured (someone did this before you got on the scene, of course), is:

```js
if (this.udid) {
  // many lines of code
} else {
  if (ver >= 5) {
    // many many lines of code
  } else {
    // 3 lines of code
  }
}
```

This gets hard to read because there are so many lines of code it can be easy to get lost and know which branch of the conditional you're in. We also lose horizontal space due to indentations. A way to solve that is:

```js
if (this.udid) {
  // many lines of code
  return;
}

if (ver < 5) {
  // 3 lines of code
  return;
}

// many lines of code
```

This "short-circuiting" kind of pattern can definitely be used to add confusion instead of take it away, but in this case I think it might make things more clear. What do you think @imurchie?